commands/message_daemon_test.go

```diff
@@ -138,3 +138,38 @@ func TestMessageSendBlockGasLimit(t *testing.T) {
 		assert.NotEmpty(t, result.Messages, "msg under the block gas limit passes validation and is run in the block")
 	})
 }
+
+func TestMessageStatus(t *testing.T) {
+	t.Parallel()
+	d := makeTestDaemonWithMinerAndStart(t)
+	defer d.ShutdownSuccess()
+
+	t.Run("queue then on chain", func(t *testing.T) {
+		assert := assert.New(t)
+
+		msg := d.RunSuccess(
+			"message", "send",
+			"--from", fixtures.TestAddresses[0],
+			"--gas-price", "0", "--gas-limit", "300",
+			"--value=1234",
+			fixtures.TestAddresses[1],
+		)
+
+		msgcid := strings.Trim(msg.ReadStdout(), "\n")
+		status := d.RunSuccess("message", "status", msgcid).ReadStdout()
+
+		assert.Contains(status, "In outbox")
+		assert.Contains(status, "In mpool")
+		assert.NotContains(status, "On chain")
+		assert.Contains(status, "1234") // the "value"
+
+		d.RunSuccess("mining once")
+
+		status = d.RunSuccess("message", "status", msgcid).ReadStdout()
+
+		assert.NotContains(status, "In outbox")
+		assert.NotContains(status, "In mpool")
+		assert.Contains(status, "On chain")
+		assert.Contains(status, "1234") // the "value"
+	})
+}
```

Comment:
blocking-ish -- we can has sad path tests?  e.g. invalid CID, or MessageFind fails

Reply:
The MessageFind not finding the message is already exercised. I added a test for a CID not found at all.